### PR TITLE
Switch to docker buildx

### PR DESCRIPTION
**Pull request by bdovaz**
In order to unlock #1553 we start with what @echoix comments: https://github.com/oxsecurity/megalinter/pull/1553#issuecomment-1236967025

### CODE REVIEW

File: .github/workflows/auto-update-linters.yml [hunk 1]
```diff
@@ -49,7 +58,7 @@ jobs:
       ###################################
       - name: Build image
         shell: bash
-        run: DOCKER_BUILDKIT=1 docker build --build-arg "BUILD_DATE=${BUILD_DATE}" --build-arg "BUILD_REVISION=auto_update_${GITHUB_SHA}" --build-arg "BUILD_VERSION=auto_update_${GITHUB_SHA}" --no-cache -t oxsecurity/megalinter:auto_update_${GITHUB_SHA} . | while read line ; do echo "$(date +'%H:%M:%S')| $line"; done;
+        run: DOCKER_BUILDKIT=1 docker buildx build --platform linux/amd64 --build-arg "BUILD_DATE=${BUILD_DATE}" --build-arg "BUILD_REVISION=auto_update_${GITHUB_SHA}" --build-arg "BUILD_VERSION=auto_update_${GITHUB_SHA}" --no-cache -t oxsecurity/megalinter:auto_update_${GITHUB_SHA} . | while read line ; do echo "$(date +'%H:%M:%S')| $line"; done;
```
Comment (echoix) [inline .github/workflows/auto-update-linters.yml:61]: I don't think the `DOCKER_BUILDKIT=1` is needed anymore, when we are using buildx directly 
```suggestion
        run: docker buildx build --platform linux/amd64 --build-arg "BUILD_DATE=${BUILD_DATE}" --build-arg "BUILD_REVISION=auto_update_${GITHUB_SHA}" --build-arg "BUILD_VERSION=auto_update_${GITHUB_SHA}" --no-cache -t oxsecurity/megalinter:auto_update_${GITHUB_SHA} . | while read line ; do echo "$(date +'%H:%M:%S')| $line"; done;
```

File: .github/workflows/deploy-DEV.yml [hunk 2]
```diff
@@ -85,7 +94,7 @@ jobs:
              MEGA_LINTER_BASE_IMAGE="oxsecurity/megalinter:${{steps.image_tag.outputs.tag}}"
              echo "Reusing previously built image as base for quick build: ${MEGA_LINTER_BASE_IMAGE}"
           fi
-          DOCKER_BUILDKIT=1 docker build --build-arg "BUILD_DATE=${BUILD_DATE}" --build-arg "BUILD_REVISION=${GITHUB_SHA}" --build-arg "BUILD_VERSION=${GITHUB_SHA}" --build-arg "MEGALINTER_BASE_IMAGE=${MEGA_LINTER_BASE_IMAGE}" --no-cache -f "Dockerfile-quick" -t oxsecurity/megalinter:${{steps.image_tag.outputs.tag}} . | while read line ; do echo "$(date +'%H:%M:%S')| $line"; done;
+          DOCKER_BUILDKIT=1 docker buildx build --platform linux/amd64 --build-arg "BUILD_DATE=${BUILD_DATE}" --build-arg "BUILD_REVISION=${GITHUB_SHA}" --build-arg "BUILD_VERSION=${GITHUB_SHA}" --build-arg "MEGALINTER_BASE_IMAGE=${MEGA_LINTER_BASE_IMAGE}" --no-cache -f "Dockerfile-quick" -t oxsecurity/megalinter:${{steps.image_tag.outputs.tag}} . | while read line ; do echo "$(date +'%H:%M:%S')| $line"; done;
```
Comment (echoix) [inline .github/workflows/deploy-DEV.yml:97]:
```suggestion
          docker buildx build --platform linux/amd64 --build-arg "BUILD_DATE=${BUILD_DATE}" --build-arg "BUILD_REVISION=${GITHUB_SHA}" --build-arg "BUILD_VERSION=${GITHUB_SHA}" --build-arg "MEGALINTER_BASE_IMAGE=${MEGA_LINTER_BASE_IMAGE}" --no-cache -f "Dockerfile-quick" -t oxsecurity/megalinter:${{steps.image_tag.outputs.tag}} . | while read line ; do echo "$(date +'%H:%M:%S')| $line"; done;
```

File: .github/workflows/deploy-DEV.yml [hunk 3]
```diff
@@ -117,7 +126,7 @@ jobs:
           !contains(github.event.head_commit.message, 'quick build')
         id: docker_build
         shell: bash
-        run: DOCKER_BUILDKIT=1 docker build --build-arg "BUILD_DATE=${BUILD_DATE}" --build-arg "BUILD_REVISION=${GITHUB_SHA}" --build-arg "BUILD_VERSION=${GITHUB_SHA}" --no-cache -t oxsecurity/megalinter:${{steps.image_tag.outputs.tag}} . | while read line ; do echo "$(date +'%H:%M:%S')| $line"; done;
+        run: DOCKER_BUILDKIT=1 docker buildx build --platform linux/amd64 --build-arg "BUILD_DATE=${BUILD_DATE}" --build-arg "BUILD_REVISION=${GITHUB_SHA}" --build-arg "BUILD_VERSION=${GITHUB_SHA}" --no-cache -t oxsecurity/megalinter:${{steps.image_tag.outputs.tag}} . | while read line ; do echo "$(date +'%H:%M:%S')| $line"; done;
```
Comment (echoix) [inline .github/workflows/deploy-DEV.yml:129]:
```suggestion
        run: docker buildx build --platform linux/amd64 --build-arg "BUILD_DATE=${BUILD_DATE}" --build-arg "BUILD_REVISION=${GITHUB_SHA}" --build-arg "BUILD_VERSION=${GITHUB_SHA}" --no-cache -t oxsecurity/megalinter:${{steps.image_tag.outputs.tag}} . | while read line ; do echo "$(date +'%H:%M:%S')| $line"; done;
```

**Comment by bdovaz** [comment]
Thanks @echoix for the suggestions.

**Comment by bdovaz** [comment]
@echoix do you know what happened in the build? It seems that the build has been done well but in the next step that makes docker run of the previously generated image, it seems that it does not find it.

**Comment by bdovaz** [comment]
@nvuillam I got this error:

image

image

**Comment by nvuillam** [comment]
Sometimes it happens, you just have to run the job again:)

**Comment by nvuillam** [comment]
@bdovaz you forgot a sh file to update  ^^

image

**Comment by bdovaz** [comment]
@nvuillam I think it is ready

**Review by nvuillam — APPROVED** [review]
Great :)
Let's see if it goes well with the deploy BETA jobs :)

**Comment by nvuillam** [comment]
thanks a lot @echoix for the help :)

**Pull request merged**